credentials/java/com/google/auth/Credentials.java

```diff
@@ -77,4 +106,20 @@
    * @throws IOException if there was an error getting up-to-date access.
    */
   public abstract void refresh() throws IOException;
+
+  /**
+   * The callback that receives the result of the asynchronous {@link #getRequestMetadata(URI,
+   * Executor, Callback)}. Exactly one method should be called.
+   */
+  public interface Callback {
```

Comment:
Would this make sense as a a top level type with a name like RequestMetadataCallback? Unless they are very obscure it be better to avoid nested types.